sdk/core/Azure.Core.TestFramework/src/HeaderRegexSanitizer.cs

```diff
@@ -8,8 +8,8 @@ public partial class HeaderRegexSanitizer
         public static HeaderRegexSanitizer CreateWithQueryParameter(string headerKey, string queryParameter, string value) =>
             new(headerKey, value)
             {
-                Regex = $@"([\x0026|&|?]{queryParameter}=)(?<group>[\w\d%]+)",
+                Regex = $@"([\x0026|&|?]{queryParameter}=)(?<group>[^&]+)",
```

Comment:
For those of us who don't speak RegEx well, a comment would be very much appreciated.